Tasks/CondaEnvironmentV1/conda_internal.ts

```diff
@@ -14,13 +14,35 @@ import { Platform } from './taskutil';
  * @returns Whether `searchDir` has a `conda` executable for `platform`.
  */
 function hasConda(searchDir: string, platform: Platform): boolean {
-    const conda = platform === Platform.Windows ?
-        path.join(searchDir, 'Scripts', 'conda.exe') :
-        path.join(searchDir, 'bin', 'conda');
+    let conda = path.join(binaryDir(searchDir, platform), 'conda');
+    if (platform === Platform.Windows) {
+        conda += '.exe';
+    }
 
     return fs.existsSync(conda) && fs.statSync(conda).isFile();
 }
 
+/**
+ * Get the platform-dependent path where binaries are located in an environment.
+ * Windows: environmentRoot\Scripts
+ * Linux / macOS: environmentRoot/bin
+ */
+function binaryDir(environmentRoot: string, platform: Platform): string {
+    if (platform === Platform.Windows) {
+        return path.join(environmentRoot, 'Scripts');
+    } else {
+        return path.join(environmentRoot, 'bin');
+    }
+}
+
+function sudo(toolPath: string, platform: Platform): ToolRunner {
+    if (platform === Platform.Windows) {
+        return task.tool('conda');
+    } else {
+        return task.tool('sudo').line('conda');
+    }
+}
```

Comment:
For cases where sudo is not installed, like for container image `ubuntu:latest`, can we run 'which sudo' to look for it, and if found use it, otherwise don't?

Reply:
Good suggestion.  I would say a few things:
1. Why doesn't `ubuntu:latest` have `sudo` installed?
2. I had originally hoped, instead of `sudo`, to create envs in a directory that the non-root user has permissions for.  Unfortunately, there's still some business with a temp file getting written to the /usr directory.  (Whenever I've installed Miniconda it's ended up in `~/miniconda`, so maybe we should install it that way on the agent.)
3. Other tasks have already taken this approach, (Use Ruby Version comes to mind) and we have a general issue with needing to run tasks as root on Linux and macOS hosted agents now.  I'd like to think through an approach that will work everywhere.

Reply:
wrt #1, I found this: https://github.com/tianon/docker-brew-ubuntu-core/issues/48
I agree this could be smarter but sounds like there is an easy workaround: `apt-get install sudo --yes`.  So I'd say it depends on how common we think this scenario will be.